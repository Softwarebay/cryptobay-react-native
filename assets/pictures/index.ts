export const BALL = require('./ball.png');
export const CONE = require('./cone.png');
export const ELLIPSES = require('./ellipses.png');
export const PAPER_PLANE = require('./paper_plane.png');
export const PROFILE = require('./profile.png');
export const SETUP = require('./setup.png');
export const SHIELD = require('./shield.png');
