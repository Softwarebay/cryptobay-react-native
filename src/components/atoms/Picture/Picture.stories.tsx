import { storiesOf } from '@storybook/react-native';
import React from 'react';

import {
  BALL,
  CONE,
  PAPER_PLANE,
  PROFILE,
  SETUP,
  SHIELD,
} from '../../../../assets/pictures';
import StorybookView from '../../../../storybook/StorybookView';

import { Picture } from './Picture';

const pictureCustomStyles = {
  width: 50,
  height: 50,
};

storiesOf('Atoms/Picture', module)
  .addDecorator(getStory => <StorybookView>{getStory()}</StorybookView>)
  .add('Ball Picture', () => <Picture pictureFile={BALL} />)
  .add('Cone Picture', () => <Picture pictureFile={CONE} />)
  .add('Paper Plane Picture', () => <Picture pictureFile={PAPER_PLANE} />)
  .add('Profile Picture', () => <Picture pictureFile={PROFILE} />)
  .add('Styled Paper Plane Picture', () => (
    <Picture pictureFile={PAPER_PLANE} style={pictureCustomStyles} />
  ))
  .add('Setup Picture', () => <Picture pictureFile={SETUP} />)
  .add('Shield Picture', () => <Picture pictureFile={SHIELD} />);
