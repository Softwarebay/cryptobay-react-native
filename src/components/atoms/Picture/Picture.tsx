import React from 'react';
import { Image, ImageSourcePropType, ImageStyle } from 'react-native';

type PictureProps = {
  pictureFile: ImageSourcePropType;
  style?: ImageStyle | ImageStyle[];
  testID?: string;
};

export const Picture: React.FC<PictureProps> = ({
  pictureFile,
  style,
  testID,
}: PictureProps) => {
  return <Image testID={testID} source={pictureFile} style={style} />;
};
