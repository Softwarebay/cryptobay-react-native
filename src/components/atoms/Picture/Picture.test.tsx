import { render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { CONE } from '../../../../assets/pictures';

import { Picture } from './Picture';

describe('Picture Atom', () => {
  test('Component Snapshot', () => {
    const tree = renderer.create(<Picture pictureFile={CONE} />);
    expect(tree).toMatchSnapshot();
  });

  test('ensure custom styling works correctly', () => {
    const customStyles = { width: 50, height: 50 };
    render(
      <Picture testID="atom-picture" pictureFile={CONE} style={customStyles} />,
    );
    expect(screen.getByTestId('atom-picture')).toHaveStyle({
      width: 50,
      height: 50,
    });
  });
});
