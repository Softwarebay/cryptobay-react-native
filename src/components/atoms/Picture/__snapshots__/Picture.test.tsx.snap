// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture Atom Component Snapshot 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../assets/pictures/cone.png",
    }
  }
/>
`;
